keps/sig-node/2371-cri-pod-container-stats/README.md

```diff
@@ -0,0 +1,416 @@
+|                          | RSSBytes             | cAdvisor               | CRI                 |
+|                          | PageFaults           | cAdvisor               | CRI                 |
+|                          | MajorPageFaults      | cAdvisor               | CRI                 |
+| AcceleratorStats         | Make                 | cAdvisor               | cAdvisor or N/A     |
+|                          | Model                | cAdvisor               | cAdvisor or N/A     |
+|                          | ID                   | cAdvisor               | cAdvisor or N/A     |
+|                          | MemoryTotal          | cAdvisor               | cAdvisor or N/A     |
+|                          | MemoryUsed           | cAdvisor               | cAdvisor or N/A     |
+|                          | DutyCycle            | cAdvisor               | cAdvisor or N/A     |
+| VolumeStats              | All Fields           | Kubelet                | Kubelet             |
+| Ephemeral Storage        | All Fields           | Kubelet                | Kubelet             |
+| Rootfs.FsStats           | AvailableBytes       | cAdvisor or N/A        | CRI or N/A          |
+|                          | CapacityBytes        | cAdvisor or N/A        | CRI or N/A          |
+|                          | UsedBytes            | CRI                    | CRI                 |
+|                          | InodesFree           | cAdvisor or N/A        | CRI or N/A          |
+|                          | Inodes               | cAdvisor or N/A        | CRI or N/A          |
+|                          | InodesUsed           | CRI                    | CRI                 |
+| UserDefinedMetrics       | All Fields           | cAdvisor               | cAdvisor or N/A     |
+
+
+## Motivation
+
+We want to avoid using cAdvisor for container & pod level stats and move metric collection to CRI runtime for the following reasons:
+
+* cAdvisor and metric dependency: CRI mission is not fully fulfilled - container runtime is not fully plugable 
+* It would be ideal to break the monolithic design of cAdvisor (i.e. cAdvisor currently needs to be aware of the underlying container runtimes) 
+* Duplicate stats are collected by both cAdvisor and the CRI runtime 
+    * Can lead to:
+        * Different information from different sources
+        * Confusion: Unclear origin of a given metric
+        * Performance degradations (increased CPU / Memory / etc) [xref][perf-issue]
+* Stats should be reported by the container runtime which knows behavior of the container/pod the best.
+* cAdvisor only supports runtimes that run processes on the host, not e.g. VM based runtime like Kata Containers
+* cAdvisor has big list of external dependencies since it needs to support multiple container runtimes itself - leads to large support surface
+
+[perf-issue]: https://github.com/kubernetes/kubernetes/issues/51798
+
+### Goals
```

Comment:
These look like implementation details, rather than Goals.  Based on my reading thus far, i'd suggest:

* Use a single source for all pod and container metrics in the Summary API to improve performance and reduce confusion.
* No breaking changes to the Summary API.
* Eliminate code dependencies on container runtime (e.g. containerd, docker, mesos) clients used by cAdvisor

Reply:
updated as suggested :)